src/common/snippets/src/pass/fq_decomposition.cpp

```diff
@@ -59,20 +82,9 @@ ov::snippets::pass::FakeQuantizeDecomposition::FakeQuantizeDecomposition() {
         if (status) {
             out_scales = calculateScales(fake_quantize_node->get_output_element_type(0), cl, ch, isc, ish, osc, osh);
         }
-        const bool do_dequantize = !(status && ((std::all_of(osc.cbegin(),
-                                                             osc.cend(),
-                                                             [](float val) {
-                                                                 return val == 1.f;
-                                                             }) &&
-                                                 std::all_of(osh.cbegin(),
-                                                             osh.cend(),
-                                                             [](float val) {
-                                                                 return val == 0.f;
-                                                             })) ||
-                                                out_scales.size() != 0));
+        const bool do_dequantize = !(status && has_zero_shifts_and_scales(cl, ish, osc, osh));
```

Comment:
Discussed offline:
1. We have to find the exact root cause of accuracy problems - the current fix enables `dequantization` for [many cases](https://github.com/openvinotoolkit/openvino/blob/9ed0fd9d4eb7be64c1e15678f8315ae66075a37f/src/common/snippets/include/snippets/pass/fq_decomposition.hpp#L38) which were efficiently executed before. Probably there some math mistake in the method `calculateScales` - need to deeper investigate
2. Please add the function tests that will reproduce accuracy problems on the both supported platforms: x64 and arm64e.

Reply:
Re-evaluated the root cause. LPT transformations produce different graphs and we have different situations on x64 and arm.